pkg/cmd/roachtest/run.go

```diff
@@ -439,6 +449,41 @@ func maybeDumpSummaryMarkdown(r *testRunner) error {
 	return nil
 }
 
+func maybeEmitDatadogEvent(
```

Comment:
Nit: It wasn't immediately obvious to me that `maybe` here implies only when a client is available. A comment on the function might be helpful.